test/fast/FastFrontendFacet.test.ts

```diff
@@ -62,8 +63,32 @@ describe('FastFrontendFacet', () => {
   });
 
   describe('emitDetailsChanged', async () => {
-    it('requires tha the caller is the diamond');
-    it('emits a DetailsChanged event with all the correct information');
+    it('requires tha the caller is the diamond', async () => {
+      const subject = frontend.emitDetailsChanged();
+      await expect(subject).to.have.been
+        .revertedWith(INTERNAL_METHOD);
+    });
+
+    it('emits a DetailsChanged event with all the correct information', async () => {
+      const frontendAsItself = await impersonateContract(frontend);
+
+      // Fire off the events but wait for the transaction.
+      const subject = frontendAsItself.emitDetailsChanged();
+
+      // Get the other details from a standard `details` function call.
+      const detailsObj = abiStructToObj(await frontend.details());
+
+      await expect(subject).to
+        .emit(frontend, 'DetailsChanged')
+        .withArgs(
+          detailsObj.memberCount,
+          detailsObj.governorCount,
+          detailsObj.totalSupply,
+          detailsObj.transferCredits,
+          detailsObj.reserveBalance,
+          BigNumber.isBigNumber /* the balance from detailsObj.ethBalance will not be correct */
```

Comment:
There's an issue here... which is why I'm pulling the predicate function in to handle this. Essentially though the `ethBalance` changes (due to the `emit`) happening I think. So, getting the `ethBalance` elsewhere and then matching on it doesn't work.